From 9a8268c4057c7887f1152af042b244c4d8e04f46 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Thu, 27 Nov 2025 16:26:36 +0800
Subject: [PATCH 244/254] mtk: hostapd: do not check ACL for link addrss

This commit reverts part of the upstream change
(https://git.w1.fi/cgit/hostap/commit/?id=d48a63d4da44c8cbd85dd409a7da68d0e2a052e2)

We want hostapd to skip ACL checks on the link address.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ieee802_11.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 4f57ada10..4bf539f72 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -3429,23 +3429,6 @@ static void handle_auth(struct hostapd_data *hapd,
 	if (res == HOSTAPD_ACL_PENDING)
 		return;
 
-#ifdef CONFIG_IEEE80211BE
-	if (mld_sta) {
-		res = ieee802_11_allowed_address(hapd, mgmt->sa,
-						 (const u8 *) mgmt, len,
-						 &rad_info);
-		if (res == HOSTAPD_ACL_REJECT) {
-			wpa_msg(hapd->msg_ctx, MSG_DEBUG,
-				"Ignore Authentication frame from " MACSTR
-				" due to ACL reject", MAC2STR(mgmt->sa));
-			resp = WLAN_STATUS_UNSPECIFIED_FAILURE;
-			goto fail;
-		}
-		if (res == HOSTAPD_ACL_PENDING)
-			return;
-	}
-#endif /* CONFIG_IEEE80211BE */
-
 #ifdef CONFIG_SAE
 	if (auth_alg == WLAN_AUTH_SAE && !from_queue &&
 	    (auth_transaction == 1 ||
-- 
2.45.2

